### PR TITLE
TLS fix for OMP clone syscall.

### PR DESCRIPTION
**Pull request by rahahahat**

### CODE REVIEW

File: src/include/simeng/arch/aarch64/Architecture.hh [hunk 1]
```diff
@@ -15,6 +15,18 @@ namespace simeng {
 namespace arch {
 namespace aarch64 {
 
+/** Enum which holds AArch64 System register tags used by SimEng. */
+enum ARM64_SYSREG_TAGS : uint16_t {
+  DCZID_EL0 = 0,
```
Comment (jj16791) [inline src/include/simeng/arch/aarch64/Architecture.hh:20]: Could we change this to have the numbering be implicit

Comment (rahahahat) [reply src/include/simeng/arch/aarch64/Architecture.hh:20]: Not possible for enums. What would be the benefit of making the numbering implicit?

Comment (FinnWilkinson) [reply src/include/simeng/arch/aarch64/Architecture.hh:20]: Implicity value allocation is possible, and also looks cleaner

File: src/lib/OS/SimOS.cc [hunk 2]
```diff
@@ -321,6 +321,9 @@ int64_t SimOS::cloneProcess(uint64_t flags, uint64_t stackPtr,
   } else if (Config::get()["Core"]["ISA"].as<std::string>() == "AArch64") {
     newProc->context_.regFile[arch::aarch64::RegisterType::GENERAL][31] = {
         stackPtr, 8};
+    newProc->context_.regFile[arch::aarch64::RegisterType::SYSTEM]
+                             [arch::aarch64::ARM64_SYSREG_TAGS::TPIDR_EL0] =
+        tls;
```
Comment (FinnWilkinson) [inline src/lib/OS/SimOS.cc:326]: Updated `TPIDR_EL0` value needs to be a reg value, i.e. `{tls, 8}`

File: src/lib/arch/aarch64/Architecture.cc [hunk 3]
```diff
@@ -28,14 +28,14 @@ Architecture::Architecture() : microDecoder_(std::make_unique<MicroDecoder>()) {
                (config["Core"]["Timer-Frequency"].as<uint32_t>() * 1e6);
 
   // Generate zero-indexed system register map
```
Comment (FinnWilkinson) [inline src/lib/arch/aarch64/Architecture.cc:30]: Update comment

**Review by FinnWilkinson — CHANGES_REQUESTED** [review]

File: src/include/simeng/arch/aarch64/Architecture.hh (hunk 1 above)
Comment (rahahahat) [reply src/include/simeng/arch/aarch64/Architecture.hh:20]: Sorry, I misunderstood what you said, I thought you meant do something similar to `systemRegisterMap_.size()`. Changing it

Comment (rahahahat) [reply src/include/simeng/arch/aarch64/Architecture.hh]: done

File: src/lib/arch/aarch64/Architecture.cc (hunk 3 above)
Comment (rahahahat) [reply src/lib/arch/aarch64/Architecture.cc]: done

**Review by jj16791 — APPROVED** [review]

**Pull request merged**